### PR TITLE
Add jq filter

### PR DESCRIPTION
**Pull request by robgmills**
Adds the ability to select content from JSON data using `jq` - a lightweight JSON processor - filter syntax.  The `jq` filters are similar to CSS or XPath selectors, with added operators to transform the data.

Addresses #592 in a different way.

### CODE REVIEW

**Comment by robgmills** [comment]
Thanks for sharing this project with us!  I'm not super well-versed in Python projects and this has been a good experience.

I can see in the Travis job how to run tests and that I have some syntax issues in the documentation.  I'll add a test or tests for this filter - is there anything in particular you want me to test for?  I'll also clean up the documentation syntax issues I introduced.

**Comment by robgmills** [comment]
These documentation driven tests are pretty cool!  I'll have to use them as inspiration later.

Sorry for the disjointed commits - I normally try to keep things cleaner.  This was a good learning experience!  I'm excited to hear your thoughts!

File: lib/urlwatch/filters.py [hunk 1]
```diff
@@ -855,3 +860,29 @@ def filter(self, data, subfilter):
...
+        except ValueError:
+            raise ValueError('The url response contained invalid JSON')
+
+        if 'query' not in subfilter:
+            raise ValueError('{} filter needs a query'.format(self.__kind__))
+
+        if jq is None:
+            raise ImportError('Please install jq')
+
+        return jq.text(subfilter['query'], json.loads(data))
```
Comment (thp) [inline lib/urlwatch/filters.py:888]: One "optimization" here could be to store the result of `json.loads(data)` in line 878, and re-use the result in line 888 to not have to re-parse the JSON data.

**Review by thp — CHANGES_REQUESTED** [review]
Nice and clean. One comment about avoiding double-parsing of JSON. Also, you could add this change to `CHANGELOG.md`.

**Comment by robgmills** [comment]
Thanks for the feedback @thp!

**Review by thp — APPROVED** [review]

**Comment by thp** [comment]
Merged, thanks!

**Comment by smithl** [comment]
Appreciate this addition very much, nice work!

**Pull request merged**